src/Microsoft.AspNet.Razor/Generator/Compiler/CodeTree/Chunks/TagHelpers/TagHelperChunk.cs

```diff
@@ -29,5 +29,10 @@ public class TagHelperChunk : ChunkBlock
         /// The HTML tag name.
         /// </summary>
         public string TagName { get; set; }
+
+        /// <summary>
+        /// Indicates whether or not the tag is self closing.
+        /// </summary>
+        public bool SelfClosing { get; set; }
```

Comment:
everywhere along this chain except in `TagHelperOutput`, `SelfClosing` should be a get-only auto-property that's set using a new constructor parameter.  this makes the property's nature much more clear.